### PR TITLE
added styles submit save to db and runcode with filepath

### PR DESCRIPTION
**Pull request by TonyDeng1997**

### CODE REVIEW

File: microservice-portal/src/main/java/com/mocking/service/UserServiceImpl.java [hunk 1]
```diff
@@ -47,5 +48,7 @@ public String findHomePathByUsername(String username) {
 	@Override
 	public String generateUserHomePath(String userLoginName) {
 		return "/tmp/"+  userLoginName + "-" + UUID.randomUUID().toString();
-	}	
+	}
```
Comment (xiaofengli) [inline microservice-portal/src/main/java/com/mocking/service/UserServiceImpl.java]: is this get or generate?

File: microservice-portal/src/main/java/com/mocking/model/AnswerSubmission.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	
+	
+	@Id
+	@Column(name = "id")
+	@GeneratedValue
+	private int id;
+	private Timestamp submission_time;
+	private String code;
+	private int user_id;
+	private int question_id;
```
Comment (xiaofengli) [inline microservice-portal/src/main/java/com/mocking/model/AnswerSubmission.java]: dont use underscore, please install sonarlint that will tell this practise is bad.

File: microservice-portal/src/main/java/com/mocking/ws/QuestionController.java [hunk 3]
```diff
@@ -120,4 +128,15 @@ public Question createQuestion(@Valid @RequestBody Question question) {
         map.put("items", questions);
         return new ResponseEntity<>(map, HttpStatus.OK);
     }
+    @ApiOperation(value = "Submit code to run test")
+    @PostMapping("/problems/submitcode")
```
Comment (xiaofengli) [inline microservice-portal/src/main/java/com/mocking/ws/QuestionController.java]: /problems/code/submit

File: microservice-portal/src/main/java/com/mocking/service/UserServiceImpl.java (hunk 1 above)
Comment (TonyDeng1997) [reply microservice-portal/src/main/java/com/mocking/service/UserServiceImpl.java]: This one is to generate

**Pull request merged**